lib/utils.ts

```diff
@@ -138,4 +138,58 @@ function checkCompatibilityWithExistingValue(value: unknown, existingValue: unkn
     };
 }
 
-export default {isEmptyObject, fastMerge, formatActionName, removeNestedNullValues, checkCompatibilityWithExistingValue};
+/**
+ * Filters an object based on a condition and an inclusion flag.
+ *
+ * @param obj - The object to filter.
+ * @param condition - The condition to apply.
+ * @param include - If true, include entries that match the condition; otherwise, exclude them.
+ * @returns The filtered object.
+ */
+function filterObject<TValue>(obj: Record<string, TValue>, condition: string | string[] | ((entry: [string, TValue]) => boolean), include: boolean): Record<string, TValue> {
+    const result: Record<string, TValue> = {};
+    const entries = Object.entries(obj);
+
+    for (let i = 0; i < entries.length; i++) {
+        const [key, value] = entries[i];
+        let shouldInclude: boolean;
+
+        if (Array.isArray(condition)) {
+            shouldInclude = condition.includes(key);
+        } else if (typeof condition === 'string') {
+            shouldInclude = key === condition;
+        } else {
+            shouldInclude = condition(entries[i]);
+        }
+
+        if (include ? shouldInclude : !shouldInclude) {
+            result[key] = value;
+        }
+    }
```

Comment:
I think that you could use `Object.fromEntries` for this, but I'm not insisting 😄 

```ts
const result = Object.fromEntries(
    entries.filter(([key, value]) => {
        let shouldInclude;

        if (Array.isArray(condition)) {
            shouldInclude = condition.includes(key);
        } else if (typeof condition === 'string') {
            shouldInclude = key === condition;
        } else {
            shouldInclude = condition([key, value]);
        }

        return include ? shouldInclude : !shouldInclude;
    })
);
```

Reply:
Yeah I chose the classic for-loop for performance reasons